File: src/console/screen/impl/DefaultScreen.php
<?php

namespace pocketcloud\cloud\console\screen\impl;

use pocketcloud\cloud\console\command\CommandManager;
use pocketcloud\cloud\console\command\sender\ConsoleCommandSender;
use pocketcloud\cloud\console\Console;
use pocketcloud\cloud\console\log\CloudLogger;
use pocketcloud\cloud\console\screen\IScreen;
use pocketcloud\cloud\util\Utils;

final class DefaultScreen extends IScreen {

    public function initialize(Console $console): void {
        $this->restorePrompt();
        $this->restoreCompletionHandler();
        $this->restoreControlCHandler();
    }

    public function tick(int $currentTick): void {}

    public function handleInput(string $input): void {
        $parts = Utils::parseQuoteAware($input);
        if (!CommandManager::getInstance()->handleInput(new ConsoleCommandSender(), $name = array_shift($parts), $parts)) {
            CloudLogger::get()->warn("§cUnknown command §8'§b" . $name . "§r§8'§c. §rView all the commands by doing §8'§bhelp§8'§r.");
        }
    }

    public function onRemove(int $currentTick): void {}
}

File: src/main/java/de/pocketcloud/cloud/console/screen/impl/DefaultScreen.java
package de.pocketcloud.cloud.console.screen.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.CloudConsole;
import de.pocketcloud.cloud.console.command.CommandManager;
import de.pocketcloud.cloud.console.command.sender.ConsoleCommandSender;
import de.pocketcloud.cloud.console.screen.Screen;

public final class DefaultScreen extends Screen {

    @Override
    public void initialize(CloudConsole console) {
        console.resetPrompt();
        console.enableCompletion();
    }

    @Override
    public void tick(long currentTick) {}

    @Override
    public void onRemove(long currentTick) {}

    @Override
    public void onCancel(long currentTick) {
        CommandManager.instance().call(new ConsoleCommandSender(), "exit -y");
    }

    @Override
    public void handleInput(String input) {
        PocketCloud.instance().commandManager().call(new ConsoleCommandSender(), input);
    }
}